src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -528,6 +533,24 @@ export class ChatWidget extends Disposable implements IChatWidget {
 	setDynamicChatTreeItemLayout(numOfChatTreeItems: number, maxHeight: number) {
 		this._dynamicMessageLayoutData = { numOfMessages: numOfChatTreeItems, maxHeight };
 		this._register(this.renderer.onDidChangeItemHeight(() => this.layoutDynamicChatTreeItemMode()));
+
+		this._register(this.tree.onDidScroll((e) => {
+			const disposable = this._register(dom.scheduleAtNextAnimationFrame(() => {
```

Comment:
This is very dangerous, as you keep adding disposables to the Disposable.store.
You should call this._unregister(disposable) when you dispose the disposable.

Also, I wonder what happens if `onDidScroll` is called multiple times before the next animation frame is scheduled.

Reply:
There isn't a `this._unregister(disposable)` 🤔 but very good call out. Refactored this to use a `MutableDisposable`

Reply:
Jeez, maybe I'm an LLM more than I thought :D